### PR TITLE
Only VEVENT for birthday calendar and default order to 1

### PR DESCRIPTION
**Pull request by tomneedham**
## Description
New calendars are given a default order of 1, to always be after the default birthday calendar.

The system birthday calendar is created with VEVENT only, not VTODO as this is not possible.

## Motivation and Context
Causes issues when the birthday calendar is used for accepting events or if TODOs are created in that calendar.

## How Has This Been Tested?
 - New install and observing the created calendars
before:
screen shot 2017-05-25 at 14 17 58
after:
screen shot 2017-05-25 at 14 43 50

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by tomneedham** [comment]
@DeepDiver1975 I can't find the code path for accepting an event from another user and where it is inserted into your calendar? Wanted to check where the logic is for choosing the calendar to insert it into.

**Comment by tomneedham** [comment]
@DeepDiver1975 hmm so I see there is already a migration to fix this - https://github.com/owncloud/core/pull/25636/files but annoyingly the fix didn't work. How often is that repair step called - as we still have corrupt birthday calendars

**Comment by DeepDiver1975** [comment]
fixes https://github.com/owncloud/core/issues/26746 ❓

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 I can't find the code path for accepting an event from another user and where it is inserted into your calendar? Wanted to check where the logic is for choosing the calendar to insert it into.

It is within the sabre dav libs:

this propfind handles the calendar selection - https://github.com/fruux/sabre-dav/blob/master/lib/CalDAV/Schedule/Plugin.php#L240
this is where the local delivery happens - https://github.com/fruux/sabre-dav/blob/master/lib/CalDAV/Schedule/Plugin.php#L444

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 hmm so I see there is already a migration to fix this - https://github.com/owncloud/core/pull/25636/files but annoyingly the fix didn't work. How often is that repair step called - as we still have corrupt birthday calendars

best would be to add a migration from my pov

**Comment by tomneedham** [comment]
> fixes #26746 ❓

yes should do - since I just push the birthday to the bottom of the users list

File: apps/dav/appinfo/Migrations/Version20170526100342.php [hunk 1]
```diff
@@ -19,44 +19,24 @@
...
 		$updated = $query->update('calendars')
 			->set('components', $query->createNamedParameter('VEVENT'))
-			->where($query->expr()->eq('uri', $query->createNamedParameter(BirthdayService::BIRTHDAY_CALENDAR_URI)))
+			->set('calendarorder', $query->createNamedParameter('100'))
+			->where($query->expr()->eq(
+				'uri',
+				$query->createNamedParameter('contact_birthdays')))
 			->execute();
-
-		$output->info("$updated birthday calendars updated.");
```
Comment (DeepDiver1975) [inline apps/dav/appinfo/Migrations/Version20170526100342.php]: lets keep the logging in here

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]

**Review by DeepDiver1975 — APPROVED** [review]

**Pull request merged**